extensions/EXT/EXT_framebuffer_blit_layers.txt

```diff
@@ -0,0 +1,95 @@
+Name
+
+    EXT_framebuffer_blit_layers
+
+Name Strings
+
+    GL_EXT_framebuffer_blit_layers
+
+Contacts
+
+    Simon Zeni, Status Holdings Ltd. <simon@zeni.ca>
+
+Contributors
+
+    Simon Zeni
+
+Status
+
+    Draft
+
+Version
+
+    Version 0.1, 2022/09/16
+
+Number
+
+    OpenGL Extension #558
+    OpenGL ES Extension #341
+
+Dependencies
+
+    OpenGL ES 3.0 is required
+
+    This extension is written against the OpenGL ES 3.2 specification
+
+Overview
+
+    This extensions defines the behaviour for copying data from one layered
+    framebuffer to another layered framebuffer.
+
+New Procedures and Functions
+
+    void BlitFramebufferLayersEXT(int srcX0, int srcY0, int srcX1, int srcY1,
+                                  int dstX0, int dstY0, int dstX1, int dstY1,
+                                  bitfield mask, enum filter);
+
+    void BlitFramebufferLayerEXT(int srcX0, int srcY0, int srcX1, int srcY1,
+                                 int srcLayer, int dstX0, int dstY0, int dstX1,
+                                 int dstY1, int dstLayer, bitfield mask,
+                                 enum filter);
+
+New Tokens
+
+    None
+
+Additions to Chapter 4 of the OpenGL 3.0 Specification (Per Fragment Operations
+and the Framebuffer)
+
+    Append the following to the section 4.3.3 (Copying Pixels)
+
+    The command
+
+        void BlitFramebufferLayersEXT(int srcX0, int srcY0, int srcX1,
+            int srcY1, int dstX0, int dstY0, int dstX1, int dstY1,
+            bitfield mask, enum filter);
+
+    extends the behaviour of the command BlitFramebuffer to all the layers of
+    the textures bound to the read framebuffer and the draw framebuffer.
+
+    Calling BlitFramebufferLayersEXT will result in an INVALID_OPERATION if
+    the read and the draw framebuffers don't have the same number of layers.
+
+    The command
+
+        void BlitFramebufferLayerEXT(int srcX0, int srcY0, int srcX1, int srcY1,
+            int srcLayer, int dstX0, int dstY0, int dstX1, int dstY1,
+            int dstLayer, bitfield mask, enum filter);
+
+    extends the behaviour of the command BlitFramebuffer to a specified layer
+    of the texture bound to the read framebuffer, and a specified layer of the
+    draw framebuffer.
+
+    Calling BlitFramebufferLayerEXT will result in an INVALID_OPERATION if
+    srcLayer or dstLayer is negative, larger than the value of 
+    MAX_ARRAY_TEXTURE_LAYERS minus one, or larger than the number of layers in
+    the attached texture.
+
+Issues
+
+    None
```

Comment:
GL_OVR_multiview defines an error condition:
--
    INVALID_FRAMEBUFFER_OPERATION is generated by commands that read from the
    framebuffer such as BlitFramebuffer, ReadPixels, CopyTexImage*, and
    CopyTexSubImage*, if the number of views in the current read framebuffer
    is greater than 1.
--
That should probably apply to BlitFramebufferLayer*EXT too?

Reply:
Good catch, I'll add it. FWIW OVR_multiview2 [1] relaxes the restrictions.

[1]: https://registry.khronos.org/OpenGL/extensions/OVR/OVR_multiview2.txt

Reply:
After reading the extension, I think `GL_OVR_multiview` refers to views as eye views, similar to what OpenXR does [1].

> The most obvious use case in this model is to support two simultaneous views: one view for each eye.

I don't think that counts framebuffer layers.

[1]: https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#view_configurations

Reply:
> Finally, multiple layers of two-dimensional array textures can be
    attached to an attachment point. Such attachments represent multiple
    views, and the corresponding attachment point is considered to be
    _multiview_.

Nevermind, Ill add it.